### PR TITLE
No Issue - Adds a debug metric controller

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt [hunk 1]
```diff
@@ -144,7 +145,37 @@ interface MetricsService {
...
 
-class Metrics(private val services: List<MetricsService>, private val isTelemetryEnabled: () -> Boolean) {
+interface MetricController {
+    fun start()
+    fun stop()
+    fun track(event: Event)
+
+    companion object {
+        fun create(services: List<MetricsService>, isTelemetryEnabled: () -> Boolean): MetricController {
+            return if (BuildConfig.TELEMETRY) return DebugMetricController()
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: Is this condition reversed? It seems like a variable named BuildConfig.TELEMETRY would be true if telemetry is switched on.

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: good catch, reversed it to make it more readable but forgot to move the objects :man_facepalming:

File: app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt [hunk 2]
```diff
@@ -127,10 +131,6 @@ class GleanMetricsService(private val context: Context) : MetricsService {
     }
 
     override fun shouldTrack(event: Event): Boolean {
-        return Settings.getInstance(context).isTelemetryEnabled && event.wrapper != null
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt]: Do we still want to track events if Telemetry is disabled, even if we're not uploading?

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt]: @colintheshots We do this check in Metrics controller before this code gets called, so it was redundant

**Pull request merged**